src/payments-processors/lnurl-payments-processor.ts

```diff
@@ -0,0 +1,69 @@
+import { AxiosInstance } from 'axios'
+import { Factory } from '../@types/base'
+
+import { CreateInvoiceRequest, GetInvoiceResponse, IPaymentsProcessor } from '../@types/clients'
+import { Invoice, InvoiceStatus, InvoiceUnit } from '../@types/invoice'
+import { createLogger } from '../factories/logger-factory'
+import { randomUUID } from 'crypto'
+import { Settings } from '../@types/settings'
+
+const debug = createLogger('alby-payments-processor')
+
+export class LnurlPaymentsProcesor implements IPaymentsProcessor {
+  public constructor(
+    private httpClient: AxiosInstance,
+    private settings: Factory<Settings>
+  ) {}
+
+  public async getInvoice(invoice: Invoice): Promise<GetInvoiceResponse> {
+    debug('get invoice: %s', invoice.id)
+
+    try {
+      const response = await this.httpClient.get(invoice.verifyURL)
+
+      return {
+        id: invoice.id,
+        status: response.data.settled ? InvoiceStatus['COMPLETED'] : InvoiceStatus['PENDING'],
+      }
+    } catch (error) {
+      console.error(`Unable to get invoice ${invoice.id}. Reason:`, error)
+
+      throw error
+    }
+  }
+
+  public async createInvoice(request: CreateInvoiceRequest): Promise<any> {
+    debug('create invoice: %o', request)
+    const {
+      amount: amountMsats,
+      description,
+      requestId,
+    } = request
+
+    try {
+      const response = await this.httpClient.get(`${this.settings().paymentsProcessors?.lnurl?.invoiceURL}/callback?amount=${amountMsats}&comment=${requestId}`)
```

Comment:
should `comment` be `request.description`?

Reply:
As commented above, we only save the description/comment in our server but don't return it in the response, so it made more sense to have the pubkey only instead of the string with relay name and npub. Will check again!